microsoft/testsuites/mdatp/mdatp.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import pathlib
+from typing import Any, Dict
+
+from assertpy import fail
+
+from lisa import (
+    Node,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    UnsupportedDistroException,
+)
+from lisa.operating_system import BSD, Linux
+from lisa.sut_orchestrator import AZURE
+from lisa.testsuite import simple_requirement
+from lisa.util.logger import Logger
+
+
+@TestSuiteMetadata(
+    area="mdatp",
+    category="functional",
+    description="""
+        Tests to check and validate mdatp config info in an Azure marketplace image.
+    """,
+)
+class MdatpSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+           Check for mdatp endpoint/cloud install, dump config info.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_os=[Linux], supported_platform_type=[AZURE]
+        ),
+    )
+    def verify_mdatp_config(self, node: Node) -> None:
+        # collect some paths before we start the test
+        local_path = pathlib.PurePath(__file__).parent.joinpath("check-mdatp.sh")
+        working_path = node.get_working_path()
+        node_script_path = working_path.joinpath("check-mdatp.sh")
+
+        # copy the bash script to the node
+        node.shell.copy(local_path=local_path, node_path=node_script_path)
+        result = node.execute(
+            cmd=f"chmod +x {str(node_script_path)}",
+            shell=True,
+            sudo=True,
+        )
+        # and run the script to check and dump defender info
+        result = node.execute(
+            cmd=f"bash +x {str(node_script_path)}",
+            shell=True,
+            sudo=True,
+            no_debug_log=True,
+        )
+        script_output = result.stdout.strip()
+        exit_code = result.exit_code
+        if exit_code != 0:
+            raise SkippedException("Did not find mdatp installation on this VM.")
+
+        node.log.info("Found mdatp installation!")
+        node.log.debug(f"Found config information: {script_output}")
+        found_onboarding_info = (
+            "/etc/opt/microsoft/mdatp/mdatp_onboard.json" in script_output
+        )
+        found_install_logs = "/var/log/microsoft/mdatp/" in script_output
+        found_mde_extension = (
+            "/var/lib/waagent/Microsoft.Azure.AzureDefenderForServers.MDE.Linux"
+            in script_output
+        )
+        error_header = ""
+        if found_onboarding_info:
+            error_header += "mdatp onboarding information is present in this image!!"
+        if found_install_logs:
+            error_header += "mdatp installation logs are present in this image!!"
+        if found_mde_extension:
+            error_header += "MDE extension data was found in /var/lib/waagent!"
+        if any([found_onboarding_info, found_install_logs, found_mde_extension]):
```

Comment:
It looks either way, the test case will be failed, if the mdatp is found.

Reply:
Yes. I don't like this solution because people could intentionally publish an image with defender on it, but not have any leaked onboarding credentials in the image. We can fix it later if that ends up being a common use case and we have tons of false positives.

Reply:
From the logic, the test case either skipped or failed. No path to be passed. The pass rate is always 0. Is that intentional?

Reply:
fixed :)

Reply:
Probability yes... See below simplified logic, it looks like the test case will fail, when mdatp is detected. It's hard to understand why, and what issues the test case is to prevent.

```python
if any([found_onboarding_info, found_install_logs, found_mde_extension]):
    fail(...
else:
    fail(...
```

Reply:
It's so you get a different error message if there's a known problem or an unknown problem. It makes it easier to filter results and jump to ones that might need fixing.

Reply:
I'll add some more comments, you're right tho :)

Reply:
commented